pkg/dackbox/indexer/lazy.go

```diff
@@ -83,6 +91,44 @@ func (li *lazyImpl) runIndexing() {
 	}
 }
 
+type hashableField struct {
+	Name           string
+	Value          []byte
+	ArrayPositions []uint64
+}
+
+type hashableDoc struct {
+	ID     string
+	Fields []hashableField
+}
+
+func (li *lazyImpl) evaluateDeduping(doc *document.Document) bool {
+	hashableDoc := hashableDoc{
+		ID:     doc.ID,
+		Fields: make([]hashableField, 0, len(doc.Fields)),
+	}
+	for _, f := range doc.Fields {
+		hashableDoc.Fields = append(hashableDoc.Fields, hashableField{
+			Name:           f.Name(),
+			Value:          f.Value(),
+			ArrayPositions: f.ArrayPositions(),
+		})
+	}
+	hashValue, err := hashstructure.Hash(hashableDoc, &hashstructure.HashOptions{
+		Hasher: li.hasher,
```

Comment:
Just double-checking that this is never called concurrently? Because it will always use the same hasher instance

Reply:
Yeah, never called concurrently. There is a single worker that consumes them